src/issue.ts

```diff
@@ -0,0 +1,51 @@
+export interface Issue {
+  id: string;
+  author: Author;
+  title: string;
+  state: State;
+  comment: CommentWithReplies;
+  discussion: CommentWithReplies[];
+  labels?: Label[];
+  timestamp: number;
+}
+
+export type State = { closed: { reason: string } } & "open";
```

Comment:
This would be an Intersection Type
https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types

It allows us to accept the "open" string but also the closed object..
It would be easier to just handle strings "open" || "closed" but I think to be able to get to the close reason makes also sense.. Maybe we could move the closed reason to another property?